zerver/views/users.py

```diff
@@ -412,7 +413,7 @@ def patch_bot_backend(
         do_update_bot_config_data(bot, config_data)
 
     if len(request.FILES) == 0:
-        pass
```

Comment:
I don't think this is a correct way to do this. We should instead call this if `avatar_source` value is sent and is equal to `AVATAR_FROM_GRAVATAR` with previous value being `AVATAR_FROM_USER`.

Reply:
addressed this in the updated PR.